docs/product-editor-development/common-tasks.md

```diff
@@ -0,0 +1,43 @@
+# Common tasks
+
+> ⚠️ **Notice:** This documentation is currently a **work in progress**. While it's open to the public for transparency and collaboration, please be aware that some sections might be incomplete or subject to change. We appreciate your patience and welcome any contributions!
+
+## Adding a new field to the product editor (end-to-end)
+
+_TODO: Describe how to add a new field, including adding it to the REST API._
+
+## Adding a group/section/field next to an existing one
+
+Here's a snippet that adds a new block to the catalog section for simple products, between the first and second fields (order 15):
+
+```php
+if ( ! function_exists( 'YOUR_PREFIX_on_block_template_register' ) ) {
+	/**
+	 * Add a new block to the template.
+	 */
+	function YOUR_PREFIX_on_block_template_register( BlockTemplateInterface $template ) {
+		if ( $template instanceof ProductFormTemplateInterface && 'simple-product' === $template->get_id() ) {
+			$section = $template->get_section_by_id( 'product-catalog-section' );
+			if ( $section !== null ) {
+				$section->add_block(
+					[
+						'id'         => 'your-prefix-id',
+						'blockName'  => 'your-block-name',
+						'order'      => 15,
+						'attributes' => [
+							'key'   => 'value',
+						],
+					]
+				);
+			}
+		}
+	}
+	add_action( 'woocommerce_block_template_register', 'YOUR_PREFIX_on_block_template_register' );
+```
```

Comment:
Since this example involves block dependency, it would be better to use the the `woocommerce_block_template_area_{template_area}_after_add_block_{block_id}` hook introduced in https://github.com/woocommerce/woocommerce/pull/40139